niworkflows/anat/mni.py

```diff
@@ -214,14 +342,102 @@ def _validate_results(self):
 
 
 def mask(in_file, mask_file, new_name):
+    """
+    Applies a binary mask to an image.
+
+    Parameters
+    ----------
+    in_file : str
+              Path to a NIfTI file.
+    mask_file : str
+                Path to a NIfTI file.
+    new_name : str
+               Path/filename for the masked output image.
+
+    Returns
+    -------
+    str
+        Absolute path of the masked output image.
+
+    Notes
+    -----
+    in_file and mask_file must be in the same
+    image space and have the same dimensions.
+    """
     import nibabel as nb
     import os
-
+    # Load the input image
     in_nii = nb.load(in_file)
+    # Load the mask image
     mask_nii = nb.load(mask_file)
+    # Set all non-mask voxels in the input file to zero.
     data = in_nii.get_data()
     data[mask_nii.get_data() == 0] = 0
+    # Save the new masked image.
     new_nii = nb.Nifti1Image(data, in_nii.affine, in_nii.header)
     new_nii.to_filename(new_name)
     return os.path.abspath(new_name)
 
+
+def create_cfm(in_file, out_path, lesion_mask, global_mask=True):
```

Comment:
I would suggest the following to make this function more "nipype -ly":

1. Change function signature to make out_path optional: 
    ```Python
    def create_cfm(in_file, lesion_mask=None, global_mask=True, out_path=None):
        ...
    ```

2. Then generate the new path automatically using nipype (preferably at the beginning of the function body):
    ```Python
    if out_path is None:
        out_path = fname_presuffix(in_file, suffix='_lesionmask',  # suffix can be _cfm if you like it better
                                   newpath=os.getcwd())
    else:
        out_path = os.path.abspath(out_path) 
    ```

    This will require the following at the top:
    ```Python
    import os
    from nipype.utils.filemanip import fname_presuffix
    ```

3. Use `out_path` whenever possible, once defined that way. A nice side effect of this suggestion is that you always use `out_path` and don't split with that saving and returning a possibly modified value. There's little chance for an error there, but better be consistent by default, right?

4. When you call `create_cfm` above, let nipype generate the name (i.e. do not set `out_path`).